Formula/petsc.rb

```diff
@@ -45,16 +45,15 @@ def install
     # Avoid references to Homebrew shims
     rm_f lib/"petsc/conf/configure-hash"
 
-    if OS.mac?
-      inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
-    elsif File.readlines("#{lib}/petsc/conf/petscvariables").grep(Superenv.shims_path.to_s).any?
+    if OS.mac? || File.foreach("#{lib}/petsc/conf/petscvariables").any? { |l| l[Superenv.shims_path.to_s] }
```

Comment:
Main differences:
- `grep(...)` doesn't seem to work with string input. Passing a regex like `grep(Regexp.new(Regexp.escape(Superenv.shims_path.to_s)))` seems to work, but ended up more lengthy
- `foreach` is more memory efficient that `readlines` or `read`, as it avoids reading entire file into memory.